test/x509_test.c

```diff
@@ -276,12 +277,126 @@ static int test_x509_revoked_delete_last_extension(void)
 
     ret = 1;
 
-err:
+ err:
     X509_REVOKED_free(rev);
     X509_EXTENSION_free(ext);
     return ret;
 }
 
+static int test_drop_empty_cert_keyids(void)
+{
+    static const unsigned char commonName[] = "test";
+    X509 *x = NULL;
+    X509_NAME *subject = NULL;
+    X509_NAME_ENTRY *name_entry = NULL;
+    X509_EXTENSION *ext = NULL;
+    X509V3_CTX ctx;
+    int ret = 0;
+
+    if (!TEST_ptr(x = X509_new())
+        || !TEST_int_eq(X509_set_version(x, X509_VERSION_3), 1)
+        || !TEST_int_eq(ASN1_INTEGER_set(X509_get_serialNumber(x), 1), 1)
+        || !TEST_ptr(subject = X509_NAME_new()))
+        goto err;
+
+    name_entry = X509_NAME_ENTRY_create_by_NID(NULL, NID_commonName,
+                                               MBSTRING_ASC, commonName, -1);
+    if (!TEST_ptr(name_entry)
+        || !TEST_int_eq(X509_NAME_add_entry(subject, name_entry, -1, 0), 1)
+        || !TEST_int_eq(X509_set_subject_name(x, subject), 1)
+        || !TEST_int_eq(X509_set_issuer_name(x, subject), 1)
+        || !TEST_ptr(X509_gmtime_adj(X509_getm_notBefore(x), 0))
+        || !TEST_ptr(X509_gmtime_adj(X509_getm_notAfter(x), 24 * 3600))
+        || !TEST_int_eq(X509_set_pubkey(x, pubkey), 1))
+        goto err;
+
+    X509V3_set_ctx(&ctx, x, x, NULL, NULL, 0);
+    if (!TEST_ptr(ext = X509V3_EXT_conf(NULL, &ctx, "subjectKeyIdentifier",
+                                        "none"))
+        || !TEST_int_eq(X509_add_ext(x, ext, -1), 1))
+        goto err;
+
+    X509_EXTENSION_free(ext);
+    if (!TEST_ptr(ext = X509V3_EXT_conf(NULL, &ctx, "authorityKeyIdentifier",
+                                        "keyid, issuer"))
+        || !TEST_int_eq(X509_add_ext(x, ext, -1), 1)
+        || !TEST_int_eq(sk_X509_EXTENSION_num(X509_get0_extensions(x)), 2)
+        || !TEST_int_gt(X509_sign(x, privkey, signmd), 0)
+        || !TEST_ptr_null(X509_get0_extensions(x)))
+        goto err;
+
+    ret = 1;
+ err:
+    X509_NAME_ENTRY_free(name_entry);
+    X509_NAME_free(subject);
+    X509_EXTENSION_free(ext);
+    X509_free(x);
+    return ret;
+}
+
+static int test_drop_empty_csr_keyids(void)
```

Comment:
This tests the degenerate case of a PKCS#10 CSR with subjectKeyIdentifier=none and authorityKeyIdentifier=keyid, issuer.
Doing this new test case at CLI level would have been much simpler, and IMO sufficient.

Other, more typical cases are not covered, neither at API level in `test/recipes/25-test_req.t‎`
Yet placing SKIDs in CSRs makes little sense to me anyway, since usually the cert producer takes care of this.
Placing AKIDs in CSRs makes no sense at all.